### PR TITLE
Add GS1 capability

### PR DESCRIPTION
**Pull request by OtherBarry**
I wanted to use this library for a project, but having to replace the "X" in isbn was annoying, and there was no way to handle GS1 codes other than GTIN-13

I've written tests for GTIN-8, GTIN-12, GTIN-13, GTIN-14, GSIN and SSCC, using the [check digit calculator on the GS1 site](https://www.gs1.org/services/check-digit-calculator).

Let me know if there's anything else that needs to be done to get this merged.

### CODE REVIEW

**Comment by harens** [comment]
Wow! Thanks so much @OtherBarry for this PR, it looks amazing 🎉 (and matches the style of the other files eerily well). I'll run the tests, and assuming they pass, I'm more than happy to merge this.

> I wanted to use this library for a project, but having to replace the "X" in isbn was annoying

Thank you for using this project. I'll push a minor release with your work so that you can use this.

File: checkdigit/gs1.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    """
+    data = cleanse(data)
+    for poss_digit in range(10):  # Brute Force the 10 options
+        option = convert(poss_digit)
+        # tests it with the generated number
+        # If this fails, the next number is tried
+        if validate(data.replace("?", option)):
+            return option
+    return "Invalid"
```
Comment (harens) [inline checkdigit/gs1.py:89]:
```
checkdigit/gs1.py           24      1      8      1    94%   89
```

Seems like we're just missing a test for this line, and then it's good to go.

Comment (OtherBarry) [reply checkdigit/gs1.py:90]: Oops, should've run coverage myself. Fixed.

Reformatted with black, should match the format requirements.

**Review by harens — APPROVED** [review]
Thanks @OtherBarry for your PR! 👍 

I'll publish a new release sometime today. Now that I know other people are using this project, I'll try and spend some more time refactoring and adding new features.

Thanks again for your pull request! 🍾

**Comment by harens** [comment]
Thanks again for your amazing work @OtherBarry! @all-contributors please add @OtherBarry for code. 🎉

**Comment by harens** [comment]
Actually, as well as that, @all-contributors please add @OtherBarry for test.

**Pull request merged**